### PR TITLE
CodeFactor

### PR DESCRIPTION
**Pull request by wlamberson1**
Resolve batch of issues from CodeFactor.io

### CODE REVIEW

**Review by jason-watkins — APPROVED** [review]

File: src/prog_models/models/pneumatic_valve.py [hunk 1]
```diff
@@ -181,14 +181,13 @@ def gas_flow(self, pIn, pOut, C, A):
 
         if pIn/pOut>=threshold:
             return C*A*pIn*sqrt(k/Z/R/T*(2/(k+1))**((k+1)/(k-1)))
-        elif pIn/pOut<threshold and pIn>=pOut:
+        if pIn/pOut<threshold and pIn>=pOut:
```
Comment (teubert) [inline src/prog_models/models/pneumatic_valve.py:184]:
```suggestion
        elif pIn>=pOut:
```
The line before checks the first part of the conditional, so we can remove that and make it an elif

File: src/prog_models/models/pneumatic_valve.py [hunk 2]
```diff
@@ -181,14 +181,13 @@ def gas_flow(self, pIn, pOut, C, A):
...
             return C*A*pIn*sqrt(2/Z/R/T*k/(k-1)*abs((pOut/pIn)**(2/k)-(pOut/pIn)**((k+1)/k)))
-        elif pOut/pIn>=threshold:
+        if pOut/pIn>=threshold:
             return -C*A*pOut*sqrt(k/Z/R/T*(2/(k+1))**((k+1)/(k-1)))
-        elif pOut/pIn<threshold and pOut>pIn:
+        if pOut/pIn<threshold and pOut>pIn:
```
Comment (teubert) [inline src/prog_models/models/pneumatic_valve.py:188]:
```suggestion
        elif pOut>pIn:
```
See above

File: src/prog_models/prognostics_model.py [hunk 3]
```diff
@@ -208,17 +208,15 @@ def __init__(self, **kwargs):
             if 'process_noise' not in self.parameters:
                 self.parameters['process_noise'] = 0.1
             else:
-                self.parameters['process_noise'] = self.parameters['process_noise'] # To force  __setitem__
+                self.parameters['process_noise'] = self.parameters['process_noise']         # To force  __setitem__
```
Comment (teubert) [inline src/prog_models/prognostics_model.py:211]: What's the story with this change?

File: src/prog_models/visualize.py [hunk 4]
```diff
@@ -106,42 +106,63 @@ def set_plot_options(opt):
...
     # Get list of options (if provided)
-    try:        opt_list = list(opt.keys())
-    except:     opt, opt_list = {}, []
+    try:
+        opt_list = list(opt.keys())
+    except Exception:
+        opt, opt_list = {}, []
     
     # Fill out all options if not provided
-    if 'figsize' not in opt_list:               opt['figsize'] = (10, 9)
```
Comment (teubert) [inline src/prog_models/visualize.py:113]: Can we replace this entire block with the following (like we do elsewhere)
```
config = {  # default configuration 
   'figsize': (10, 9),
   ...  # Other defaults
}
config.update(opt_list) 
```

File: src/prog_models/visualize.py [hunk 5]
```diff
@@ -175,23 +196,39 @@ def set_legend_options(leg_opt, s_names):
     | print(leg_opt['fancybox']) # False
     | print(leg_opt['facecolor']) # 'w'
     """
-    try:        leg_list = list(leg_opt.keys())  # Check whether a dictionary has been provided. If not, initialize the dictionary leg_opt as empty
```
Comment (teubert) [inline src/prog_models/visualize.py:178]: Same note as above- replace with dict of defaults and use update to merge

File: src/prog_models/visualize.py [hunk 6]
```diff
@@ -442,20 +492,18 @@ def set_labels(ax, opt, series_names, axis='all'):
         xlabel    = extract_option(opt['xlabel'], idx, series_names)
         xtick     = extract_option(opt['xticks'], idx, series_names)
         ax.set_xlabel(xlabel)
-        if xtick:       # if xtick options are passed, add them to the axis
+        if xtick:                   # if xtick options are passed, add them to the axis
```
Comment (teubert) [inline src/prog_models/visualize.py:495]: What's going on here?

File: tests/__init__.py [hunk 7]
```diff
@@ -1,4 +1,4 @@
 # Copyright © 2021 United States Government as represented by the Administrator of the
 # National Aeronautics and Space Administration.  All Rights Reserved.
 
-__all__ = ['test_base_models', 'test_battery', 'test_examples', 'test_centrifugal_pumps', 'test_pneumatic_valve', 'visualize']
```
Comment (teubert) [inline tests/__init__.py:4]: Why are these removed?

**Review by teubert — CHANGES_REQUESTED** [review]
Some great changes! I left a few comments to resolve

**Comment by teubert** [comment]
> Some great changes! I left a few comments to resolve

Since it's been merged, my comments can be resolved in another PR

File: tests/__init__.py (hunk 7 above)
Comment (wlamberson1) [reply tests/__init__.py:4]: These were removed because they were never defined. CodeFactor didnt like it. Would you like me to add them back?

File: src/prog_models/prognostics_model.py (hunk 3 above)
Comment (wlamberson1) [reply src/prog_models/prognostics_model.py:211]: This was to improve readability. It made it even with another related comment

File: tests/__init__.py (hunk 7 above)
Comment (teubert) [reply tests/__init__.py:4]: Nope, you're right. I wonder what happened to them.
